geti_sdk/rest_clients/credit_system_client.py

```diff
@@ -0,0 +1,119 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import logging
+from typing import List, Optional, Union
+
+from geti_sdk.data_models import Subscription
+from geti_sdk.data_models.job import Job, JobCost
+from geti_sdk.http_session import GetiSession
+from geti_sdk.utils.job_helpers import get_job_by_id
+from geti_sdk.utils.serialization_helpers import deserialize_dictionary
+from geti_sdk.utils.workspace_helpers import get_default_workspace_id
+
+
+def allow_supported(func):
+    """
+    Decorate the class methods to allow them to run only if the Credit System is supported.
+    """
+
+    def wrapper(instance, *args, **kwargs):
+        if instance._is_supported:
+            return func(instance, *args, **kwargs)
+        else:
+            logging.warning(
+                "Credit System is not supported by the Intel Geti Platform."
+            )
+            return None
+
+    return wrapper
+
+
+class CreditSystemClient:
+    """
+    Class to work with credits in Intel Geti.
+    """
+
+    def __init__(self, session: GetiSession, workspace_id: Optional[str] = None):
+        self.session = session
+        if workspace_id is not None:
+            self.workspace_id = workspace_id
+        else:
+            self.workspace_id = get_default_workspace_id(self.session)
+        # Make sure the Intel Geti Platform supports Credit System.
+        self._is_supported = self.is_supported()
+
+    def is_supported(self) -> bool:
+        """
+        Check if the Intel Geti Platform supports Credit system.
+
+        :return: True if the Credit System is supported, False otherwise.
+        """
+        r = self.session.get_rest_response(
+            url=self.session.base_url + "balance",
+            method="GET",
+        )
+        if isinstance(r, dict):
+            # If the Platform responds with the information about the available subscriptions,
+            # then it supports Credit System. Session will decode the message into a `dict`
+            return True
+        else:
+            # In case the server returns an empty 200 message, it means the Credit System is not supported.
+            # The session would return a Response object.
+            return False
+
+    @allow_supported
+    def get_balance(self) -> Optional[int]:
+        """
+        Get the current credit balance in the workspace.
+
+        :return: The available credit balance in the workspace.
+        """
+        response = self.session.get_rest_response(
+            url=self.session.base_url + "balance",
+            method="GET",
+        )
+        return response.get("available", None)
```

Comment:
Should we maybe include also the possibility to see historical values, and also `incoming` balance values?
cc @mpryahin

![image](https://github.com/user-attachments/assets/18aa6aa3-7265-4479-a4df-3fde52053db1)

Reply:
added the query parameter